tools/cdk-release/lib/conventional-commits.ts

```diff
@@ -0,0 +1,82 @@
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const conventionalCommitsParser = require('conventional-commits-parser');
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const gitRawCommits = require('git-raw-commits');
+
+/**
+ * The optional notes in the commit message.
+ * Today, the only notes are 'BREAKING CHANGES'.
+ */
+export interface ConventionalCommitNote {
+  /** Today, always 'BREAKING CHANGE'. */
+  readonly title: string;
+
+  /** The body of the note. */
+  readonly text: string;
+}
+
+/** For now, only needed for unit tests. */
+export interface ConventionalCommitReference {
+}
+
+export interface ConventionalCommit {
+  /** The type of the commit ('feat', 'fix', etc.). */
+  readonly type: string;
+
+  /** The optional scope of the change ('core', 'aws-s3', 's3', etc.). */
+  readonly scope?: string;
+
+  /** The subject is the remaining part of the first line without 'type' and 'scope'. */
+  readonly subject: string;
+
+  /** The header is the entire first line of the commit
+   * (<type>(<scope>): <subject>).
+   */
+  readonly header: string;
+
+  /**
+   * The optional notes in the commit message.
+   * Today, the only notes are 'BREAKING CHANGES'.
+   */
+  readonly notes: ConventionalCommitNote[];
+
+  /**
+   * References inside the commit body
+   * (for example, to issues or Pull Requests that this commit is linked to).
+   */
+  readonly references: ConventionalCommitReference[];
+}
+
+/**
+ * Returns a list of all Conventional Commits in the Git repository since the tag for `currentVersion`.
+ * The commits will be sorted in chronologically descending order
+ * (that is, later/newer commits will be earlier in the array).
+ *
+ * @param currentVersion the string representing the current version,
+ *   will be used to limit the returned commits to only those added after that version's release
+ */
+export async function getConventionalCommitsFromGitHistory(currentVersion: string): Promise<ConventionalCommit[]> {
+  const ret = new Array<any>();
+  return new Promise((resolve, reject) => {
+    const conventionalCommitsStream = gitRawCommits({
+      format: '%B%n-hash-%n%H',
+      // our tags have the 'v' prefix
+      from: `v${currentVersion}`,
```

Comment:
Does this mean from PREVIOUS VERSION to HEAD ?

Reply:
What if we have multiple versions in the repo? Then this won't be a single version for all packages, but it will be package-dependent which version we need?

Reply:
The experimental modules will use the V2 release tag for determining the commits that happened since the last release.

Reply:
Doesn't it make more sense to make the git tag (or honestly, any kind of `refspec`) a parameter of this function? Feels like incorporating the convention from version-number-to-tag here is at the wrong level of abstraction.

Reply:
Sure. Changed.